src/Commands/Generators/NewShieldOauthGenerator.php

```diff
@@ -0,0 +1,188 @@
+{
+    use GeneratorTrait;
+
+    /**
+     * The Command's Group
+     *
+     * @var string
+     */
+    protected $group = 'ShieldOAuth';
+
+    /**
+     * The Command's Name
+     *
+     * @var string
+     */
+    protected $name = 'make:oauth';
+
+    /**
+     * The Command's Description
+     *
+     * @var string
+     */
+    protected $description = 'Adds new OAuth to Shield OAuth.';
+
+    /**
+     * The Command's Usage
+     *
+     * @var string
+     */
+    protected $usage = 'make:oauth [<name>] [options]';
+
+    /**
+     * The Command's Arguments
+     *
+     * @var array<string, string>
+     */
+    protected $arguments = [
+        'name' => 'The name of the new OAuth without the `OAuth` suffix. The `OAuth` suffix will be automatically added for you.',
+    ];
+
+    /**
+     * The Command's Options
+     *
+     * @var array<string, string>
+     */
+    protected $options = [
+        '--force' => 'Force overwrite existing file.',
+    ];
+
+    /**
+     * Path to the app folder
+     *
+     * @var string
+     */
+    protected $distPath = APPPATH;
+
+    /**
+     * The ContentReplacer class
+     */
+    protected ContentReplacer $replacer;
+
+    /**
+     * Actually execute a command.
+     */
+    public function run(array $params): int
+    {
+        // The name param is required
+        if (empty($params[0])) {
+            CLI::error('Please enter the name of the OAuth.', 'light_gray', 'red');
+
+            return 1;
+        }
+
+        $this->replacer = new ContentReplacer();
+
+        $this->component     = 'Library';
+        $this->directory     = 'Libraries\ShieldOAuth';
+        $this->template      = 'newoauth.tpl.php';
+        $this->classNameLang = 'CLI.generator.className.library';
+        $this->setHasClassName(false);
+
+        // Disable the suffix option
+        $this->setEnabledSuffixing(false);
+
+        // The proper class name should contain the `OAuth` suffix if it doesn't exist
+        $class = str_ireplace('oauth', 'OAuth', $params[0]);
+        if (strpos($class, 'OAuth') === false) {
+            $class .= 'OAuth';
+        }
+
+        $params[0] = $class;
+
+        // ShieldOAuthConfig file must be present
+        if ($this->updateConfig($class) !== 0) {
+            CLI::error('ShieldOAuthConfig.php does not exist. Please run `php spark make:oauthconfig` first.', 'light_gray', 'red');
+
+            return 1;
+        }
+        // @TODO execute() is deprecated in CI v4.3.0.
+        $this->execute($params); // @phpstan-ignore-line suppress deprecated error.
+
+        return 0;
+    }
+
+    /**
+     * Add new OAuth to ShieldOAuthConfig.
+     *
+     * @param string $className
+     */
+    private function updateConfig($className): int
+    {
+        // Check that the config file exists
+        $file = 'Config/ShieldOAuthConfig.php';
+        if (! is_file($this->distPath . $file)) {
+            return 1;
+        }
+
+        // Add the code
```

Comment:
Honestly, I got through it, but I'm back up again.
Using code(codeKey,codeName) does not look good and does not have good readability.
Please use another word(maybe OauthKey,OauthName is better.).